src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -0,0 +1,303 @@
+            {
+                // Use Property first
+                attr = options.GetAttributes<JsonEnumerableConverterAttribute>(propertyInfo).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+            }
+
+            if (attr == null)
+            {
+                // Then class type
+                attr = options.GetAttributes<JsonEnumerableConverterAttribute>(parentClassType, inherit: true).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+
+                if (attr == null)
+                {
+                    // Then declaring assembly
+                    attr = options.GetAttributes<JsonEnumerableConverterAttribute>(parentClassType.Assembly).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+
+                    if (attr == null)
+                    {
+                        // Then global
+                        attr = options.GetAttributes<JsonEnumerableConverterAttribute>(JsonSerializerOptions.GlobalAttributesProvider).Where(a => a.EnumerableType == enumerableType).FirstOrDefault();
+
+                        // Then default
+                        if (enumerableType == typeof(Array))
+                        {
+                            attr = s_arrayConverterAttibute;
+                        }
+                    }
+                }
+            }
+
+            return attr?.CreateConverter();
+        }
+
+        public static JsonValueConverter<TProperty> GetPropertyValueConverter<TProperty>(
+            Type parentClassType,
+            PropertyInfo propertyInfo,
+            Type propertyType,
+            JsonSerializerOptions options)
+        {
+            Type propertyTypeNullableStripped;
+
+            JsonValueConverterAttribute attr = GetPropertyValueConverterInternal(parentClassType, propertyInfo, options, propertyType);
+            if (attr != null)
+            {
+                return attr.GetConverter<TProperty>();
+            }
+
+            bool isNullable = propertyType.IsGenericType && propertyType.GetGenericTypeDefinition() == typeof(Nullable<>);
+            if (isNullable)
+            {
+                propertyTypeNullableStripped = Nullable.GetUnderlyingType(propertyType);
+                attr = GetPropertyValueConverterInternal(parentClassType, propertyInfo, options, propertyTypeNullableStripped);
+                if (attr != null)
+                {
+                    return attr.GetConverter<TProperty>();
+                }
+            }
+            else
+            {
+                propertyTypeNullableStripped = propertyType;
+            }
+
+            // For Enums, support both the type Enum plus strongly-typed Enums.
+            if (propertyTypeNullableStripped.IsEnum || propertyTypeNullableStripped == typeof(Enum))
+            {
+                attr = GetPropertyValueConverterInternal(parentClassType, propertyInfo, options, typeof(Enum));
+                if (attr == null)
```

Comment:
Consider changing to `return (attr ?? s_enumConverterAttribute).GetConverter<TProperty>();`?

Reply:
Thanks. I cleaned this up a bit to remove the extra null check.